lib/streamlit/web/server/server.py

```diff
@@ -396,9 +396,9 @@ def browser_is_connected(self) -> bool:
 
     @property
     def is_running_hello(self) -> bool:
-        from streamlit.hello import Hello
+        from streamlit.hello import streamlit_app
 
-        return self._main_script_path == Hello.__file__
+        return self._main_script_path == streamlit_app.__file__
```

Comment:
nit: maybe also do a manual check if the hello app flag is set in the page profile when you run it via `streamlit hello`

Reply:
Sorry, I'm having trouble understanding the sentence. From what I see a Page Profile does not understand `streamlit hello`. Are you saying:

1. Don't send a Page Profile for Streamlit Hello
2. Extend Page Profile with a boolean whether it's `streamlit hello`
3. Something else.

Reply:
The page profile has this `isHello` flag which is important that it is set to `true` when Streamlit is started via `streamlit hello`: 

<img width="620" alt="image" src="https://github.com/streamlit/streamlit/assets/2852129/c71f9b1c-9bfa-4ff3-9ed6-198dad21b35c">

It would be good to just double check that this is still the case with the new hello app + the subpages of the hello app. But I assume that is the case.

Reply:
btw. it is a bit strange why this isn't part of the `PageProfile` proto and is instead set on the frontend. I don't think there is a specific reason why it needs to be set on the frontend instead of the backend.

Reply:
Ah! It's not in the Page Profile itself, but the event attaches extra info in metrics. I've confirmed it!

<img width="1166" alt="Screenshot 2024-06-03 at 10 00 47 AM" src="https://github.com/streamlit/streamlit/assets/69432/cd2758d2-7f82-481a-9319-f25edea52255">

Reply:
isHello is sent on any metrics call, and is also used for not displaying the deploy button.